# Python As a Calculator

shift + enter

Python Shell


In [1]:
1+2

3


![Getting Started](./Arithmetic_Ops.jpg)

In [2]:
(3*4)/(2**2 + 4/2)

2.0

Compute 3 divided by 4, then multiply the result by 2, and then raise the result to the 3rd power.

In [3]:
3/4

0.75

In [4]:
_*2

1.5

In [5]:
_**3

3.375

Python has many basic arithmetic functions like sin, cos, tan, asin, acos, atan, exp, log, log10 and sqrt stored in a module (We will explain it later in this chapter) called math.

In [6]:
import math

TIP! In Jupyter notebook and Ipython, you can have a quick view of what’s in the module by type the module name + dot + TAB. Furthermore, if you type the first few letters of the function and press TAB, it could automatically complete the function for you, the so called - TAB completion.

In [7]:
math.cosh

<function math.cosh(x, /)>

TRY IT! Find the square root of 4.

In [8]:
math.sqrt(4)

2.0

TRY IT! Compute the 𝑠𝑖𝑛(𝜋/2)

In [9]:
math.sin(math.pi/2)

1.0

Python will compose functions as you would expect, with the innermost function being executed first. The same holds true for function calls that are composed with arithmetic operations.

TRY IT! Compute 𝑒^𝑙𝑜𝑔10.

In [11]:
math.exp(math.log(10))

10.000000000000002

TIP! You can see the result above should be 10, but it is showing 10.000000000000002, this is due to Python’s number approximation, which we will learn more in chapter 9.

TRY IT! Compute 𝑒^3/4

In [12]:
math.exp(3/4)

2.117000016612675

Many times, when you use a function in Python, you want to get some help on the context of the function. In Ipython or Jupyter notebook you can view the description of any function by typing function?, that is the question mark is a shortcut to get help. If you see a function you are unfamiliar with, it is good practice to use the question mark before you asking your instructors what a specific function does.

WARNING! For some functions, the help file can be extremely complicated and wordy, even for simple functions. In these cases, do not be afraid to ask your instructor for help.

In [13]:
math.factorial?

Signature: math.factorial(x, /)
Docstring:
Find x!.

Raise a ValueError if x is negative or non-integral.
Type:      builtin_function_or_method

Python will raise an ZeroDivisionError when you have expression 1/0, which is infinity, to remind you.

In [14]:
1/0

ZeroDivisionError: division by zero

You can type math.inf at the command prompt to denote infinity or math.nan to denote something that is not a number that you wish to be handled as a number. If this is confusing, this distinction can be skipped for now; it will be explained more clearly when it becomes important. Finally, Python can also handle the imaginary number.

TRY IT! 1/∞
, and ∞∗2
 to verify that Python handles infinity as you would expect.

In [16]:

1/math.inf


0.0

In [17]:
math.inf * 2

inf

TRY IT! Compute ∞/∞

In [18]:
math.inf / math.inf

nan

TRY IT! Compute sum 2 + 5i

In [19]:
2+5j

(2+5j)

Note that, in Python imaginary part is using j instead of i to represent.

Another way to represent complex number in Python is to use the complex function.

In [20]:
complex(2,5)

(2+5j)

In [21]:
complex(2)

(2+0j)

Python can also handle scientific notation using the letter e between two numbers. For example, 1𝑒6=1000000
 and 1𝑒−3=0.001
.

TRY IT! Compute the number of seconds in 3 years using scientific notation.

In [22]:
3e0*3.65e2*2.4e1*3.6e3

94608000.0

TIP! Every time when we type the function in math module, we always type math.function_name. Alternatively, there is a simpler way, for example, if we want to use sin and log from math module, we could import them this way: from math import sin, log. Then you all you need to do when using these functions is using them directly, for example, sin(20) or log(10).

